File: WorkoutTracker/src/app/workout-progress/current-user-progress.ts
import { ModelWeight } from "../shared/models/model-weight";
import { WorkoutStatus } from "./workout-status";

export interface CurrentUserProgress {
    workoutStatus: WorkoutStatus;

    countOfWorkoutDays: number;
    firstWorkoutDate: Date|null;
    lastWorkoutDate: Date|null;

    currentWorkoutStrikeDays: number;
    currentBodyWeight: ModelWeight;
}

File: WorkoutTrackerAPI/Data/Models/ProgressModels/CurrentUserProgress.cs
namespace WorkoutTrackerAPI.Data.Models.ProgressModels;

public class CurrentUserProgress
{
    public WorkoutStatus? WorkoutStatus { get; set; }
    
    public DateTime? FirstWorkoutDate { get; set; }
    public DateTime? LastWorkoutDate { get; set; }
    public int CountOfWorkoutDays { get; set; }

    public int CurrentWorkoutStrikeDays { get; set; }
    public ModelWeight CurrentBodyWeight { get; set; }
}
